### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by amrc-benmorrow**
Initial implementation of a Git server.

* Serve Git repos integrated with the F+ auth framework.
* Allow repos to be created and deleted via an API.
* Support groups of repos to allow for more flexible permission management.

### CODE REVIEW

File: PROCESS.md
Comment (AlexGodbehere) [inline PROCESS.md:1]: I propose that we keep all documentation one place (namely the EDO), not in the Git repo. Thoughts?

File: bin/convert-dumps.js
Comment (AlexGodbehere) [inline bin/convert-dumps.js:1]: Could you add a comment explaining exactly what this does and why it's needed? I have a decent idea but may be difficult to follow for the non-initiated.

File: dumps/git-auth.json
Comment (AlexGodbehere) [inline dumps/git-auth.json:1]: Might be worth adding some comments to these to make it easier to determine what they are?

File: dumps/git-auth.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+service: !u FP.Service.Authentication
```
Comment (AlexGodbehere) [inline dumps/git-auth.yaml:1]: What does `!u` do?

File: PROCESS.md
Comment (amrc-benmorrow) [reply PROCESS.md:1]: Yes, this file should be removed.

File: bin/convert-dumps.js
Comment (amrc-benmorrow) [reply bin/convert-dumps.js:1]: Yes, that would be a good idea. I would like to move this functionality into `@amrc-factoryplus/utilities`, or (better) into a separate `@amrc-factoryplus/convert-service-dumps` module that can be a `devDependency`, but I can document this version here for now.

File: dumps/git-auth.json
Comment (amrc-benmorrow) [reply dumps/git-auth.json:1]: You can't put comments in JSON. This is one of the reasons I was generating it from YAML instead.

File: dumps/git-auth.yaml (hunk 1 above)
Comment (amrc-benmorrow) [reply dumps/git-auth.yaml:1]: `!u` is a YAML custom data type marker. In this case it is expanded by `convert-dumps.js` into a UUID pulled either from `@amrc-factoryplus/utilities` or from `lib/uuids.js` as appropriate.

I'm trying to get to the point of having a SSoT for these UUIDs. It is concerning me that we are copy-pasting them all the time.

File: dumps/git-auth.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# This file is processed by `npm run convert-dumps` into
+# `git-auth.json`. That file then needs to be loaded into the Auth
+# service.
```
Comment (AlexGodbehere) [inline dumps/git-auth.yaml:3]:
```suggestion
# component.
```

File: dumps/git-configdb.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# This file is processed by `npm run convert-dumps` into
+# `git-configdb.json`. That file then needs to be loaded into the
+# ConfigDB.
```
Comment (AlexGodbehere) [inline dumps/git-configdb.yaml:3]:
```suggestion
# Configuration Store component.
```

**Review by AlexGodbehere — APPROVED** [review]

**Pull request merged**